src/Data/Lens/Types.purs

```diff
@@ -27,21 +27,45 @@ import Data.Profunctor.Choice (class Choice)
 import Data.Profunctor.Closed (class Closed)
 import Data.Profunctor.Strong (class Strong)
 
--- | A general-purpose Data.Lens.
-type Optic p s t a b = p a b -> p s t
-type Optic' p s a = Optic p s s a a
+-- | Given a type whose "focus element" can always be retrieved,
+-- | a lens provides a convenient way to view, get, and transform
+-- | that element. 
+-- | 
+-- | `_2` is a Tuple-specific `Lens` available from `Data.Lens`, so:
+-- | ```purescript
+-- | > import Data.Lens
+-- | > over _2 String.length $ Tuple "ignore" "four"
+-- | (Tuple "ignore" 4)
+-- | ```
+type Lens s t a b = forall p. Strong p => Optic p s t a b
+
+-- | `Lens` allows `set` to change the type of the focus. Often, a
+-- | particular lens won't do that. This type alias declares that `set`
+-- | only changes values, not types.
```

Comment:
I think this comment is a little weird. It's hard to see how the description relates to the type - I'd probably at least start off with something like "A more specialized form of `Lens`..." so writing about what it does do, rather than explaining it in terms of what `Lens'` does that this can't.

I guess I'd come at it from the angle that the docs for each type and value should be self contained; no assumption that the file is being read from top to bottom (since when I look up things in pursuit I tend to just read the definition, not the whole module that the definition resides with in - also editor tooling shows the docs for one thing at a time).

Reply:
Changed.